lib/xnvme_be_linux_async_liburing.c

```diff
@@ -392,6 +403,47 @@ xnvme_be_linux_liburing_cmd_iov(struct xnvme_cmd_ctx *ctx, struct iovec *dvec, s
 
 	return 0;
 }
+
+int
+xnvme_be_linux_liburing_get_completion_fd(struct xnvme_queue *queue)
+{
+	struct xnvme_queue_liburing *q = (struct xnvme_queue_liburing *)queue;
+	int efd, err;
+
+	if (q->efd != -1) {
+		return q->efd;
+	}
+
+	if (q->batching) {
+		XNVME_DEBUG("Completion FD requested while batching is on, submitting any "
```

Comment:
I still don't think xnvme_queue_get_completion_fd() should be submitting I/O. It would be preferable to add a guard in xnvme_queue.c that returns EBUSY, to avoid redundancy and ensure consistent behavior across all backends.

Reply:
> I still don't think xnvme_queue_get_completion_fd() should be submitting I/O. It would be preferable to add a guard in xnvme_queue.c that returns EBUSY, to avoid redundancy and ensure consistent behavior across all backends.

Adding it to xnvme_queue.c might not be the way to go. However, adding a guard is srill preferable.

Reply:
Since the user must handle "spurious" events, an alternative approach would be to write to the eventfd before returning it.